src/Core/Services/GraphQLRequestValidator.cs

```diff
@@ -0,0 +1,538 @@
+                    HashSet<string>? derivedColumnsForEntity, columnsToBeDerivedFromEntity;
+
+                    // When the current entity is a referenced entity, there will be no corresponding entry for the relationship
+                    // in the derivedColumnsForRelationships dictionary.
+                    columnsDerivedForRelationships.TryGetValue(relationshipName, out derivedColumnsForEntity);
+
+                    // When the current entity is a referencing entity, there will be no corresponding entry for the relationship
+                    // in the columnsToBeDerivedFromRelationships dictionary.
+                    columnsToBeDerivedFromRelationships.TryGetValue(relationshipName, out columnsToBeDerivedFromEntity);
+                    ValidateGraphQLValueNode(
+                        schema: schemaObject.Fields[relationshipName],
+                        entityName: targetEntityName,
+                        context: context,
+                        parameters: fieldDetails.Item1,
+                        runtimeConfig: runtimeConfig,
+                        columnsDerivedFromParentEntity: derivedColumnsForEntity ?? new(),
+                        columnsToBeDerivedFromEntity: columnsToBeDerivedFromEntity ?? new(),
+                        nestingLevel: nestingLevel,
+                        parentEntityName: entityName,
+                        sqlMetadataProviderFactory: sqlMetadataProviderFactory);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Helper method to validate that the referencing columns are not included in the input data for the target (referencing) entity -
+        /// because the value for such referencing columns is derived from the insertion in the source (referenced) entity.
+        /// In case when a value for referencing column is also specified in the referencing entity, there can be two conflicting sources of truth,
+        /// which we don't want to allow. In such a case, we throw an appropriate exception.
+        /// </summary>
+        /// <param name="columnsInTargetEntity">Columns in the target (referencing) entity.</param>
+        /// <param name="derivedColumnsFromSourceEntity">Foreign key columns to be derived from the source (referenced) entity.</param>
+        /// <param name="nestingLevel">Current depth of nesting in the multiple-create GraphQL request.</param>
+        /// <param name="targetEntityName">Name of the target entity.</param>
+        /// <param name="metadataProvider">Metadata provider.</param>
+        private static void ValidateAbsenceOfReferencingColumnsInTargetEntity(
+            Dictionary<string, string> columnsInTargetEntity,
+            HashSet<string> derivedColumnsFromSourceEntity,
+            int nestingLevel,
+            string targetEntityName,
+            ISqlMetadataProvider metadataProvider)
+        {
+            foreach (string derivedColumnFromParentEntity in derivedColumnsFromSourceEntity)
```

Comment:
Source means the upper level entity in the mutation. It cannot be child. Child is always the target entity in a relationship. Basically child = target. I will update referenced to child by target entity since this seems to be causing confusion.